### PR TITLE
add property ownerPickerMode to TechDocsIndexPage and DefaultApiExplorerPage

### PR DESCRIPTION
**Pull request by benjidotsh**
## Hey, I just made a Pull Request!

I noticed that `TechDocsIndexPage` and `DefaultApiExplorerPage` didn't contain the `ownerPickerMode` property like `CatalogIndexPage` does, so this PR adds this property to the other pages as well.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by benjidotsh** [comment]
@Rugvip @camilaibs can someone please take a look at this? 😇

File: .changeset/gentle-baboons-peel.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-techdocs': minor
+---
+
+Added property ownerPickerMode to TechDocsIndexPage
```
Comment (vinzscam) [inline .changeset/gentle-baboons-peel.md:5]:
```suggestion
`TechDocsIndexPage` now accepts an optional `ownerPickerMode` for toggling the behavior of the `EntityOwnerPicker`, exposing a new mode `<TechDocsIndexPage ownerPickerMode="all" />` particularly suitable for larger catalogs. In this new mode, `EntityOwnerPicker` will display all the users and groups present in the catalog.
```

File: .changeset/gentle-baboons-peel.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-techdocs': minor
```
Comment (vinzscam) [inline .changeset/gentle-baboons-peel.md:2]:
```suggestion
'@backstage/plugin-techdocs': patch
```

File: .changeset/tiny-pandas-return.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-api-docs': patch
+---
+
+Added property ownerPickerMode to DefaultApiExplorerPage
```
Comment (vinzscam) [inline .changeset/tiny-pandas-return.md:5]:
```suggestion
`DefaultApiExplorerPage` now accepts an optional `ownerPickerMode` for toggling the behavior of the `EntityOwnerPicker`, exposing a new mode `<DefaultApiExplorerPage ownerPickerMode="all" />` particularly suitable for larger catalogs. In this new mode, `EntityOwnerPicker` will display all the users and groups present in the catalog.
```

**Review by vinzscam — CHANGES_REQUESTED** [review]
Thank you @benjidotsh, the changes make sense 🙏

I've just suggested to expand the changesets to match the catalog ones

**Review by vinzscam — APPROVED** [review]

**Review by squid-ney — APPROVED** [review]

**Comment by squid-ney** [comment]
Thanks for your contribution!

**Pull request merged**